react/components/ProductDescription/components/GradientCollapse/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Transition } from 'react-transition-group'
+import debounce from 'debounce'
+import { FormattedMessage } from 'react-intl'
+import './global.css'
+
+class GradientCollapse extends Component {
+  constructor(props){
+    super(props)
+    this.state = { showCollapse: true, collapsed: true, maxHeight: 'auto' }
+
+    this.wrapper = React.createRef()
+  }
+
+  calcMaxHeight = () => {
+    const {collapseHeight} = this.props
+    if (this.wrapper.scrollHeight > collapseHeight) {
+      const maxHeight = this.wrapper.scrollHeight + 60
+      this.setState({showCollapse: true, maxHeight})
+    } else this.setState({showCollapse: false, maxHeight: 'auto'})
+  }
+
+  debouncedCalcMaxHeight = debounce(this.calcMaxHeight, 500)
+
+  componentDidMount(){
+    window.addEventListener('resize', this.debouncedCalcMaxHeight)
+    this.calcMaxHeight()
+  }
+
+  componentWillUnmount(){
+    window.removeEventListener('resize', this.debouncedCalcMaxHeight)
+  }
+
+
+  render(){
+    const {children, collapseHeight} = this.props
+    const {collapsed, showCollapse, maxHeight} = this.state
+    const height = showCollapse && collapsed ? collapseHeight : maxHeight
+    const transitionTime = 600
+    const fadeOutTime = 400
+    const transitionStyle = (transitionTime) => ({transition: `${transitionTime}ms ease-in-out`})
```

Comment:
This function could be out of the class or at least out of render method since is just a specific healper function